api/app/src/shared/date.ts

```diff
@@ -31,3 +31,7 @@ export const toISODate = (unixTime: number): string => {
 export const toISODateTime = (unixTime: number): string => {
   return dayjs.unix(unixTime).toISOString();
 };
+
+export function parseISODate(date?: string): string | undefined {
+  return dayjs(date, ISO_DATE, true).isValid() ? date : undefined;
+}
```

Comment:
Ideally this would be a validator, so we can throw a BAD_REQUEST in case the param is set incorrectly - so the user gets a deterministic behavior... using `undefined` from an invalid date is non-deterministic.

We can defs create a ticket for it, though. I have some code we can use as a starting point.

Reply:
It's a good point, I was thinking about this as well - will address now